internal/migrate/stage_contract.go

```diff
@@ -66,94 +73,251 @@ func stageContract(
 		return nil, err
 	}
 
-	contractName := args[0]
-	contract, err := state.Contracts().ByName(contractName)
+	// Validate command arguments
+	optionCount := boolCount(stageContractflags.All, len(stageContractflags.Accounts) > 0, len(args) > 0)
+	if optionCount > 1 {
+		return nil, fmt.Errorf("only one of --all, --account, or contract names can be provided")
+	} else if optionCount == 0 {
+		return nil, fmt.Errorf("at least one of --all, --account, or contract names must be provided")
+	}
+
+	// Stage based on flags
+	var v stagingValidator
+	if !stageContractflags.SkipValidation {
+		v = newStagingValidator(flow)
+	}
+	s := newStagingService(flow, state, logger, v, promptStagingUnvalidatedContracts(logger))
+
+	if stageContractflags.All {
+		return stageAll(s, state, flow)
+	}
+
+	if len(stageContractflags.Accounts) > 0 {
+		return stageByAccountNames(s, state, flow, stageContractflags.Accounts)
+	}
+
+	return stageByContractNames(s, state, flow, args)
+}
+
+func promptStagingUnvalidatedContracts(logger output.Logger) func(validatorError *stagingValidatorError) bool {
+	return func(validatorError *stagingValidatorError) bool {
+		infoMessage := strings.Builder{}
+
+		infoMessage.WriteString("Preliminary validation could not be performed on the following contracts:\n")
+		missingDependencyErrors := validatorError.MissingDependencyErrors()
+		for deployLocation := range missingDependencyErrors {
+			infoMessage.WriteString(fmt.Sprintf("  - %s\n", deployLocation))
+		}
+
+		infoMessage.WriteString("\nThese contracts depend on the following contracts which have not been staged yet:\n")
+		missingDependencies := validatorError.MissingDependencies()
+		for _, depLocation := range missingDependencies {
+			infoMessage.WriteString(fmt.Sprintf("  - %s\n", depLocation))
+		}
+
+		infoMessage.WriteString("\nYou may still stage your contract, however it will be unable to be migrated until the missing contracts are staged by their respective owners.  It is important to monitor the status of your contract using the `flow migrate is-validated` command\n")
+		logger.Error(infoMessage.String())
+
+		continuePrompt := promptui.Select{
```

Comment:
There is a `GenericBoolPrompt` we should probably start using now. Although I wish we could cherry pick the new prompt changes from the other branch and convert this to using the new library, but I don't want to add more work here. Should be easy enough to change later. Might want to use the `GenericBoolPrompt` though.